### PR TITLE
Cleanup and unit test 

### PR DESCRIPTION
**Pull request by anfernee**
- Remove legacy/unused code
- Reorder imports
- Add unit test for agentclient

### CODE REVIEW

File: pkg/agent/agentclient/client_test.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	// \r\n
+	// Hello, client
+	headAndBody := strings.Split(string(data), "\r\n")
+	if body := headAndBody[len(headAndBody)-1]; body != expectedBody {
+		t.Errorf("expect body %v; got %v", expectedBody, body)
+	}
+}
+
+// stream implements AgentService_ConnectClient
+type stream struct {
```
Comment (cheftako) [inline pkg/agent/agentclient/client_test.go]: Maybe name this as testStream? fakeStream?

File: pkg/agent/agentclient/client_test.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	defer close(stopCh)
+
+	// Start test http server
+	expectedBody := "Hello, client"
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprint(w, expectedBody)
+	}))
+	defer ts.Close()
+
+	// Send DIAL_REQ packet
```
Comment (cheftako) [inline pkg/agent/agentclient/client_test.go]: Simulate sending KAS DIAL_REQ from the Proxy to the AgentClient by sending appropriate packet

File: pkg/agent/server/server.go [hunk 3]
```diff
@@ -1,179 +0,0 @@
-/*
-Copyright 2019 The Kubernetes Authors.
```
Comment (cheftako) [inline pkg/agent/server/server.go]: I know the deleted "legacy" code was actually legacy. This is also legacy?

Comment (anfernee) [reply pkg/agent/server/server.go]: Yes it is legacy. This is a one-hop proxy server without the dialing back agent.

**Pull request merged**